source/simulator/assemblers/newton_stokes.cc

```diff
@@ -0,0 +1,677 @@
+      const double theta = parameters.newton_theta;
+
+      for (unsigned int q=0; q<n_q_points; ++q)
+        {
+          for (unsigned int i=0, i_stokes=0; i_stokes<stokes_dofs_per_cell; /*increment at end of loop*/)
+            {
+              if (introspection.is_stokes_component(fe.system_to_component_index(i).first))
+                {
+                  scratch.phi_u[i_stokes] = scratch.finite_element_values[introspection.extractors.velocities].value (i,q);
+                  scratch.phi_p[i_stokes] = scratch.finite_element_values[introspection.extractors.pressure].value (i, q);
+                  scratch.grads_phi_u[i_stokes] = scratch.finite_element_values[introspection.extractors.velocities].symmetric_gradient(i,q);
+                  scratch.div_phi_u[i_stokes]   = scratch.finite_element_values[introspection.extractors.velocities].divergence (i, q);
+
+                  ++i_stokes;
+                }
+              ++i;
+            }
+
+
+          // Viscosity scalar
+          const double eta = scratch.material_model_outputs.viscosities[q];
+
+          const SymmetricTensor<4,dim> &stress_strain_director =
+            scratch.material_model_outputs.stress_strain_directors[q];
+          const bool use_tensor = (stress_strain_director !=  dealii::identity_tensor<dim> ());
+          const SymmetricTensor<2,dim> strain_rate = scratch.material_model_inputs.strain_rate[q];
+          const double pressure = scratch.material_model_inputs.pressure[q];
+          const double velocity_divergence = scratch.velocity_divergence[q];
+
+          const Tensor<1,dim>
+          gravity = this->get_gravity_model().gravity_vector (scratch.finite_element_values.quadrature_point(q));
+
+          const double density = scratch.material_model_outputs.densities[q];
+
+          const double JxW = scratch.finite_element_values.JxW(q);
+          if (theta == 0)
+            {
+              for (unsigned int i=0; i<stokes_dofs_per_cell; ++i)
+                {
+                  data.local_rhs(i) -= (eta * 2.0 * (scratch.grads_phi_u[i] * strain_rate)
+                                        - (scratch.div_phi_u[i] * pressure)
+                                        - (pressure_scaling * scratch.phi_p[i] * velocity_divergence)
+                                        -(density * gravity * scratch.phi_u[i]))
+                                       * JxW;
+
+                  if (assemble_newton_stokes_matrix)
+                    for (unsigned int j=0; j<stokes_dofs_per_cell; ++j)
+                      {
+                        data.local_matrix(i,j) += ( /*(use_tensor ?
+                                                     eta * 2.0 * (scratch.grads_phi_u[i] * stress_strain_director * scratch.grads_phi_u[j])
+                                                     :*/
+                                                    eta * 2.0 * (scratch.grads_phi_u[i] * scratch.grads_phi_u[j])//)
```

Comment:
So anisotropic viscosities are currently not supported in the Newton solver? That is not a problem with me, but then instead of commenting the code, please remove it and add an assert above that `use_tensor` is false.